.github/workflows/ci.yaml

```diff
@@ -43,17 +43,47 @@ jobs:
             fi
           done < files.txt
 
- ### web ci pipeline ###
+  ### web ci pipeline ###
   web:
     name: web-ci-pipeline
     needs: check
     if: needs.check.outputs.run_web == 'true'
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x, 20.x]
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - name: Install node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: |
+          cd web/
+          npm install
+      # - name: Build application
+      #   run: |
+      #     cd web/
+      #     npm run build
+      - name: Lint with ESLint
+        run: |
+          cd web/
+          npm run lint
+      - name: Run unit tests with coverage
+        run: |
+          cd web/
+          npm run test
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v3
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        with:
+          directory: web/cobertura-coverage.xml
```

Comment:
`Codecov` is unable to find this directory, you can try to change this to `./web`